Automattic/jetpack
# Protect: record admin page-view and get security from footer

This PR records two events:

* `jetpack_protect_admin_page_view`
* `jetpack_protect_footer_get_security_link_click`

For this purpose, it introduces the `useAnalyticsTracks()` custom hook. The original idea was to add this hook to the js-packages/analytics pkg but some dependencies conflicts make us hold on and add it straight to this plugin.

Fixes #

#### Changes proposed in this Pull Request:

* Protect: Add useAnalyticsTracks() custom hook
* Protect: Record admin page-view and get security from footer

#### Jetpack product discussion


p1651180674864999-slack-C02TQF5VAJD
p1651228540992269-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


Yes:

* `jetpack_protect_admin_page_view` when the Protect admin page shows.
* `jetpack_protect_footer_get_security_link_click` by clicking on the `Get Jetpack Security` Footer button.

#### Testing instructions:






* Set debug to show `dops` events:

```
localStorage.setItem( 'debug', 'dops:analytics*' );
```
* Confirm you see the `jetpack_protect_admin_page_view` there when the Protect admin page shows.
* Confirm you see the `jetpack_protect_footer_get_security_link_click` by clicking on the `Get Jetpack Security` Footer button.
